superset/views/utils.py

```diff
@@ -212,14 +213,20 @@ def get_time_range_endpoints(
     Get the slice aware time range endpoints from the form-data falling back to the SQL
     database specific definition or default if not defined.
 
-    For SIP-15 all new slices use the [start, end) interval which is consistent with the
-    native Druid connector.
+    When SIP-15 is enabled all slices and will the [start, end) interval. If the grace
+    period is defined and has ended all slices will adhere to the [start, end) interval.
 
     :param form_data: The form-data
     :param slc: The chart
     :returns: The time range endpoints tuple
     """
 
+    if (
+        app.config["SIP_15_GRACE_PERIOD_END"]
```

Comment:
this is saying that if you don't define a grace period, then sip 15 will never be enabled right? Maybe we should default to using the sip 15 logic when a grace period isn't specified?

Reply:
No. The SIP-15 check is upstream of the call to `get_time_range_endpoints`. If no grace period is explicitly defined the grace period is indefinite. I'll add more documentation to the config.